src/app_modes/topic_managment.rs

```diff
@@ -0,0 +1,365 @@
+        let mut rng = rand::thread_rng();
+        for topic in self.selected_topics.items.iter() {
+            match topic[1].clone().as_ref() {
+                "sensor_msgs/LaserScan" => config.laser_topics.push(
+                    ListenerConfigColor{
+                        topic: topic[0].clone(),
+                        color: ConfigColor{r:rng.gen_range(0..255), g:rng.gen_range(0..255), b:rng.gen_range(0..255)},
+                    }
+                ),
+                "visualization_msgs/MarkerArray" => config.marker_array_topics.push(
+                    ListenerConfig {
+                        topic: topic[0].clone(),
+                    }
+                ),
+                "visualization_msgs/Marker" => config.marker_topics.push(
+                    ListenerConfig {
+                        topic: topic[0].clone(),
+                    }
+                ),
+                "geometry_msgs/PoseStamped" => config.pose_stamped_topics.push(
+                    PoseListenerConfig{
+                        topic: topic[0].clone(),
+                        color: ConfigColor{r:rng.gen_range(0..255), g:rng.gen_range(0..255), b:rng.gen_range(0..255)},
+                        length: 0.2,
+                        style: "axis".to_string(),
+                    }
+                ),
+                "geometry_msgs/PoseArray" => config.pose_array_topics.push(
+                    PoseListenerConfig{
+                        topic: topic[0].clone(),
+                        color: ConfigColor{r:rng.gen_range(0..255), g:rng.gen_range(0..255), b:rng.gen_range(0..255)},
+                        length: 0.2,
+                        style: "axis".to_string(),
+                    }
+                ),
+                "nav_msgs/Path" => config.path_topics.push(
+                    PoseListenerConfig{
+                        topic: topic[0].clone(),
+                        color: ConfigColor{r:rng.gen_range(0..255), g:rng.gen_range(0..255), b:rng.gen_range(0..255)},
+                        length: 0.2,
+                        style: "axis".to_string(),
+                    }
+                ),
+                _ => (),
+            }
+        }
+
+        // Store and exit termviz
+        let _ = confy::store("termviz", "termviz", &(config));
+    }
+
+}
+
+impl<B: Backend> BaseMode<B> for TopicManager {}
+
+impl AppMode for TopicManager {
+    fn run(&mut self) {}
+    fn reset(&mut self) {}
+    fn get_description(&self) -> Vec<String> {
+        vec!["Topic manager can enable and disable displayed topics".to_string()]
+    }
+
+    fn handle_input(&mut self, input: &String) {
+        if self.selection_mode {
+            match input.as_str() {
+                input::UP => self.availible_topics.previous(),
+                input::DOWN => self.availible_topics.next(),
+                input::RIGHT => self.shift_active_element_right(),
+                input::ROTATE_RIGHT => self.selection_mode = false,
```

Comment:
switching between the modes does not work for me

Reply:
It works for me: there is just no visual feedback.
@MaxiMaerz I would make the cursor (highlighted topic) "disappear" (i.e. not bold anymore) from the list that is not active at the moment. And highlight the first entry by default if none is selected (the user needs to press w or s to have any visual feedback at the beginning.